src/components/ModalWithForm.js

```diff
@@ -6,6 +6,10 @@ export default class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
   }
+  open() {
+    this._modalForm.querySelector(".modal__button").innerText = "Save";
+    super.open();
+  }
   _getInputValues() {
     const _modalInputs = [...this._modalForm.querySelectorAll(".modal__input")];
     const inputValues = {};
```

Comment:
Please, delete ` this.close();` from line 27 in `ModalWithForm`

* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well